biocrnpyler/dna_part_promoter.py

```diff
@@ -78,24 +82,18 @@ def update_reactions(self):
                         component = self, part_id = self.name, complex = None,
                         transcript = self.transcript, protein = self.get_protein_for_expression())
         return reactions
-    def update_component(self,dna,rnas,proteins,mypos = None):
+
+    def update_component(self,internal_species=None,**keywords):
         """returns a copy of this component, except with the proper fields updated"""
-        if(dna.material_type == "rna"):
-            #Promoters only work with DNA
+        from .dna_construct import DNA_construct, RNA_construct
```

Comment:
please move the package import to the top of the file!

Reply:
@zoltuz This is necessary to avoid circular import loops. (DNA_construct.py depends on dna_part_promoter.py and visa-versa). If you know a better way to solve this, let us know!